### DISCUSSION TITLE
Implement into GridCal

### DISCUSSION THREAD

**Discussion by SanPen** [RAMP-project/RAMP#154] [Q&A]
Hi, 

I've seen this project and I'd like to use it in a load design wizard in (GridCal)[https://github.com/SanPen/GridCal]

For that I'd like to know a couple of things:
- Can it generate deterministic time series?
- Can it be "hacked" to simulate industrial loads and cities?

BR,
Santiago

**Comment by Slbalderrama** [comment]
Hello, 

for your first question:
The model's idea is to create plausible stochastic loads. Under this idea, each time you run the model, you would get different loads but with a recognizable pattern. You can validate those loads against real ones to ensure the data you are using is correct.

For the second question:

The model's idea is to create loads from the bottom up. Some papers have reproduced the demands of small communities. In this context, it is probably easier to reproduce industrial loads as it would be possible to quantify the number of pieces of equipment and the parameters needed to reproduce them in RAMP. In the case of cities, it might be more complicated.

**Reply by Bachibouzouk** [reply]
Hello @SanPen ,

Thank you for your interest in RAMP!

Since the latest version the load profiles can be made deterministic thanks to the random seed. Of you set it the stochastic profiles will always be the same. I don't know if it might be useful to your use case.

I agree with @Slbalderrama and I think you could simulate industrial loads if you can model them using the parameters of the Appliance class. What do you have as input data to simulate your industrial 
load profile? 

For the cities you could simulate one by categorizing households and public facilities by creating several User instances and assigning a number of user in each instance. As @Slbalderrama mentionned, this approach might be trickier for big  inhomogenous cities. However the case study of El Espino gave very nice results.

**Comment by SanPen** [comment]
Hi, 

Thank you for the answers. In general I have no knowledge of any particular element.
As main developer of an electrical calculation software, I'm looking for a way to create load scenarios from macro parameters.

I thought this library could be of use to build from the ground up, instead of going the ML route that may not be useful to produce never-seen-before scenarios. As you point out, maybe by "hacking" some of the existing clases I could get what I want.

As per the profile question; What I need is a single time series. I don't mind it being random in nature, but I need an array not a matrix of N possible scenarios. It's just that.

Also, if you happen to know any other resource where I could find what I'm looking for that would be very useful.

**Reply by Slbalderrama** [reply]
First comment: The purpose of RAMP is to generate load time series from the bottom-up, so your appreciation is correct. I actually have created some unseen demand scenarios for my PhD thesis for rural communities in Bolivia. It is not necessary to hack the software, just to change some parameters or new appliances or user profiles (as defined in RAMP). https://orbi.uliege.be/handle/2268/267954

Second comment: you can obtain a single demand for the community/village etc, that is not a problem.

Third comment: We have finish this paper in which we have made a state of the art with other possibilities: https://joss.theoj.org/papers/10.21105/joss.06418?fbclid=IwZXh0bgNhZW0CMTEAAR2HR7X4ucujKQRMQNBJjEM8CSncZVAzKwgDl2Lo4T2z1L6bOykmq2SkMck_aem_FrBRD1fhGgBdypCEPLC_Pg
